### ISSUE TITLE
Wrong error message for re-exporting primitive types

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#36168]
```ts
export { number };
```

**Expected**: Cannot export 'number'. Only local declarations can be exported from a module.

**Actual**: **No error message**

**Comment by andrewbranch** [comment]
@DanielRosenwasser uhm, looks like you do get an error? https://www.typescriptlang.org/play/#code/KYDwDg9gTgLgBAbzgOwK4FsBGwpwL5A

Which means we can fix this alongside #36167 for 3.8.
